src/core/core.controller.js

```diff
@@ -31,10 +31,31 @@ module.exports = function(Chart) {
 		data.datasets = data.datasets || [];
 		data.labels = data.labels || [];
 
+
+		// Merge options for mixed charts
+		// If there is at least one 'bar' chart, main type is set to 'bar'
```

Comment:
Can you expand the comment to explain why `If there is at least one 'bar' chart, main type is set to 'bar'`

One thing I'm wondering about is if there's a more generic way to do this. I'm worried that this won't work if you want to mix a financial chart and line chart for example (https://github.com/chartjs/chartjs-chart-financial)

Reply:
I share the same concerns: wouldn't be better to add support for bar specific options (scale) at the dataset level and avoid touching the options logic? Actually, I think that can cause breaking changes, especially [the logic that merges all dataset defaults on top of each others](/chartjs/Chart.js/pull/5134/files#diff-de7249441169d0e646c53d8220e8b0c5R55). I'm not even sure it works great since the last dataset (options) overwrites the previous ones.

For example, with the following types: `line` > `scatter`, the value of `showLines` will be `false` but `scatter` > `line` generates `showLines: true` (not consistent). Same for `line` > `bubble`, the tooltip options will break for the line.

Reply:
I agree that we should try to have all options supported at the dataset level